### ISSUE TITLE
Adding parameter with duplicate name should throw ModelError

### ISSUE THREAD

**Issue by georgemhall** [GillesPy2/GillesPy2#33]
I was copy-pasting some parameters and forgot to change the name argument. The same thing happened for Species before - it just needs to be applied to Parameter as well (and presumably rate_rule?)

While fixing, maybe look at consistency of error messages. The "is a duplicate" error message is worded differently for reactions and species.

**Comment by Thisisnotdalton** [comment]
This issue should be fixed in the [model_sanitization branch](https://github.com/GillesPy2/GillesPy2/tree/model_sanitization).

**Comment by briandrawert** [comment]
Can we add a unit test of this?  If done, please run and then close this issue.

**Comment by georgemhall** [comment]
I'm working on adding a test to check for thrown errors but I am getting a lot of import errors when trying to run the coverage module:

```
$ coverage run unittests/test_simple_model.py
Traceback (most recent call last):
  File "unittests/test_simple_model.py", line 6, in <module>
    from gillespy2.core import Model, Species, Reaction, Parameter, RateRule
  File "../gillespy2\__init__.py", line 1, in <module>
    from gillespy2.core import *
  File "../gillespy2\core\__init__.py", line 2, in <module>
    from gillespy2.core import gillespyError
  File "../gillespy2\core\gillespy2.py", line 16, in <module>
    from gillespy2.core.gillespySolver import GillesPySolver
ImportError: No module named core.gillespySolver
```

However, that issue doesn't happen when running it directly:

```$ py unittests/test_simple_model.py
2018-11-14 15:51:36,728 - root - WARNING -  Unable to use Cython optimized SSA: No module named 'pyximport'. The performance of this package can be significantly increased if you install Cython.
.F.F........................
======================================================================
FAIL: test_addingMultipleSameParameter_ThrowsError (__main__.TestSimpleModel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "unittests/test_simple_model.py", line 67, in test_addingMultipleSameParameter_ThrowsError
    self.model.add_parameter([k1,k2])
AssertionError: ModelError not raised

======================================================================
FAIL: test_addingSameParameter_ThrowsError (__main__.TestSimpleModel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "unittests/test_simple_model.py", line 60, in test_addingSameParameter_ThrowsError
    self.model.add_parameter(k1)
AssertionError: ModelError not raised

----------------------------------------------------------------------
Ran 28 tests in 0.067s

```

**Comment by briandrawert** [comment]
Made a comment in #42 that might fix the above error

**Comment by briandrawert** [comment]
addressed by #35
